Tools/clinic/requirements-dev.txt

```diff
@@ -0,0 +1,2 @@
+# Requirements file for external linters and checks we run on Tools/clinic/ in CI
+mypy==1.2.0
```

Comment:
We're deliberately pinning to an older version here, in order to test that dependabot updates are working on this file, as per @hugovk's suggestion in https://github.com/python/cpython/pull/104421#discussion_r1192399303